### PR TITLE
chore: remove geo and notifications code from ui-react

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
- Remove geo and notifications code from `ui-react`
- Remove unused imports in `ui-react` (like maplibre related dependencies)
- Remove geo related styles from `ui-react`
- Update e2e and unit test github workflows to only build `react-core-notifications` for related packages like `react-notifications` and `react-native`.

#### Description of how you validated changes
- unit and e2e tests

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/reusable-e2e.yml [hunk 1]
```diff
@@ -154,10 +154,6 @@ jobs:
         if: ${{ matrix.package == 'react' }}
         run: yarn react-core build
 
-      - name: Build react-core-notifications package
```
Comment (reesscot) [inline .github/workflows/reusable-e2e.yml:157]: Can you explain why this is no longer needed?

File: .github/workflows/reusable-unit.yml [hunk 2]
```diff
@@ -91,10 +91,7 @@ jobs:
 
       - name: Build react-core-notifications package
         if: |
-          matrix.package == 'react' ||
           matrix.package == 'react-native' ||
-          matrix.package == 'react-storage' ||
-          matrix.package == 'react-liveness' ||
```
Comment (reesscot) [inline .github/workflows/reusable-unit.yml:97]: Was this ever needed? Doesn't seem related

File: packages/react/package.json [hunk 3]
```diff
@@ -59,13 +58,8 @@
     "classnames": "2.3.1",
     "deepmerge": "4.2.2",
     "lodash": "4.17.21",
-    "mapbox-gl": "1.13.1",
```
Comment (reesscot) [inline packages/react/package.json:62]: 💯

File: packages/react/package.json [hunk 4]
```diff
@@ -119,12 +111,6 @@
       "import": "{ Authenticator }",
       "limit": "120 kB"
     },
-    {
```
Comment (reesscot) [inline packages/react/package.json:122]: We need to update our canaries to run the size check on the new packages. Have we done that?

File: packages/react/src/styles.css [hunk 5]
```diff
@@ -1,4 +1 @@
 @import '../../ui/dist/styles.css';
-@import '@maplibre/maplibre-gl-geocoder/dist/maplibre-gl-geocoder.css';
```
Comment (reesscot) [inline packages/react/src/styles.css:2]: 💯

File: .github/workflows/reusable-e2e.yml (hunk 1 above)
Comment (sreeramsama) [reply .github/workflows/reusable-e2e.yml:157]: We do not need to build `react-core-notifications` for `ui-react` anymore as we moved inappmessaging code out and do not have any code depending on `react-core-notifications` now.

Earlier, inappmessaging code in `ui-react` depended on `react-core-notifications`.

File: .github/workflows/reusable-unit.yml (hunk 2 above)
Comment (sreeramsama) [reply .github/workflows/reusable-unit.yml:97]: It is kind of a dependency chain.
Inappmessaging code in `ui-react` depended on `react-core-notifications`.
`react-storage` depends on `ui-react`, which depended on `react-core-notifications`.
`react-liveness` depends on `ui-react`, which depended on `react-core-notifications`.

We can remove `react`, `react-storage` and `react-liveness` from this condition now as inappmessaging code is being moved out of `react`, so `react` and other `react-*` packages do not need `react-core-notifications`.

We have inappmessaging code in `react-notifications` and `react-native`, so we want to build `react-core-notifications` for those packages.

Comment (reesscot) [reply .github/workflows/reusable-unit.yml:97]: OK, that makes more sense, thanks for clarifying!

**Review by reesscot — APPROVED** [review]
LGTM, thanks for the explanations!

File: .github/workflows/reusable-e2e.yml (hunk 1 above)
Comment (sreeramsama) [reply .github/workflows/reusable-e2e.yml]: Added this back as we need to build `react-core-notifications` for example app to use.

**Review by ioanabrooks — APPROVED** [review]

File: packages/react/package.json (hunk 3 above)
Comment (zchenwei) [reply packages/react/package.json:62]: We finally drop those dependencies!

File: packages/react/package.json (hunk 4 above)
Comment (sreeramsama) [reply packages/react/package.json:122]: Will have a PR to `main` for that
